src/js/platform/browser/game_analytics.js

```diff
@@ -103,13 +102,15 @@ export class ShapezGameAnalytics extends GameAnalyticsInterface {
                 body: JSON.stringify(data),
             })
                 .then(res => {
+                    clearTimeout(timeout);
                     if (!res.ok || res.status !== 200) {
-                        throw new Error("Fetch error: Bad status " + res.status);
+                        reject("Fetch error: Bad status " + res.status);
+                    } else {
+                        resolve(res.json());
                     }
-                    return res;
                 })
-                .then(res => res.json()),
-        ]);
+                .catch(reject);
```

Comment:
If the fetch fails, then the timeout will not be cleared - I'm not sure if thats good practice?

Reply:
Reject being called a second time doesn't do anything, but I have added a `clearTimeout` and fixed the resolve.